pwndbg/commands/asm.py

```diff
@@ -0,0 +1,75 @@
+import argparse
+
+import pwn
+import pwnlib.context
+
+import pwndbg.commands
+from pwndbg.color import message
+
+parser = argparse.ArgumentParser(description="Assemble shellcode into bytes")
+
+parser.add_argument(
+    "-f", "--format", default="hex", choices=["hex", "string"], type=str, help="Output format"
+)
+
+parser.add_argument(
+    "--arch",
+    default=pwnlib.context.context.arch,
+    choices=pwnlib.context.context.architectures.keys(),
+    type=str,
+    help="Target architecture",
+)
+
+parser.add_argument(
+    "-v",
+    "--avoid",
+    action="append",
+    help="Encode the shellcode to avoid the listed bytes (provided as hex)",
+)
+
+parser.add_argument(
+    "-n",
+    "--newline",
+    dest="avoid",
+    action="append_const",
+    const="0a",
+    help="Encode the shellcode to avoid newlines",
+)
+
+parser.add_argument(
+    "-z",
+    "--zero",
+    dest="avoid",
+    action="append_const",
+    const="00",
+    help="Encode the shellcode to avoid NULL bytes",
+)
+
+input_group = parser.add_mutually_exclusive_group(required=True)
+
+input_group.add_argument(
+    "shellcode", default=[], nargs="*", type=str, help="Assembler code to assemble"
+)
+
+input_group.add_argument("-i", "--infile", default=None, type=str, help="Specify input file")
+
+
+@pwndbg.commands.ArgparsedCommand(parser, command_name="asm")
+def asm(shellcode, format, arch, avoid, infile):
+    if infile:
+        print(message.warn("Going to read from file: " + infile))
+        with open(infile) as file:
+            shellcode = [file.read()]
+
+    assembly = pwnlib.asm.asm(" ".join(shellcode), arch=arch)
```

Comment:
```suggestion
    bits_for_arch = pwnlib.context.context.architectures.get(arch, {}).get('bits')
    assembly = pwnlib.asm.asm(" ".join(shellcode), arch=arch, bits=bits_for_arch)
```

You need to do something like this since we are updating both Pwnlib's `context.arch` and `context.bits` in one of our hooks. As a result, since bits are set explicitly and you do `asm --arch=i386 nop` when you are debugging an x86-64 (amd64) binary the bits are still set to `'64'` and we would get:
```
Exception occurred: asm: Invalid arch/bits combination: i386/64 (<class 'AttributeError'>)
```

Reply:
Thx for information :) Please verify